iina/QuickSettingViewController.swift

```diff
@@ -825,26 +845,110 @@ class QuickSettingViewController: NSViewController, NSTableViewDataSource, NSTab
-      audioEqSlider6.doubleValue,
-      audioEqSlider7.doubleValue,
-      audioEqSlider8.doubleValue,
-      audioEqSlider9.doubleValue,
-      audioEqSlider10.doubleValue,
-      ])
-  }
-
-  @IBAction func resetAudioEqAction(_ sender: AnyObject) {
-    player.removeAudioEqFilter()
-    updateAudioEqState()
+  func applyEQ(_ profile: EQProfile) {
+    zip(eqSliders, profile.gains).forEach { (slider, gain) in
+      slider.doubleValue = gain
+    }
+    player.setAudioEq(fromGains: profile.gains)
+  }
+
+  private func makeProfileNameValidationAlert(isNewProfile: Bool) -> NSAlert {
+    let panel = NSAlert()
+    let textInput = NSTextField(frame: NSRect(x: 0, y: 0, width: 200, height: 24))
+    panel.accessoryView = textInput
+    let firstButton = panel.addButton(withTitle: NSLocalizedString("general.ok", comment: "OK"))
+    firstButton.isEnabled = false
+    panel.addButton(withTitle: NSLocalizedString("general.cancel", comment: "Cancel"))
+    let key = isNewProfile ? "eq.new_profile" : "eq.rename"
+    panel.messageText = NSLocalizedString(key, comment: key)
+    let nameList = eqPopUpButton.itemArray.filter{ $0.tag == 1 || $0.tag == 0 }.map{ $0.title }
+    NotificationCenter.default.addObserver(forName: NSControl.textDidChangeNotification, object: textInput, queue: .main) { _ in
+      let input = textInput.stringValue
+      if input.isEmpty {
+        firstButton.isEnabled = false
+        return
+      }
+      if nameList.contains( where: { $0 == input } ) {
+        firstButton.isEnabled = false
+      } else {
+        firstButton.isEnabled = true
+        self.inputString = input
+      }
+    }
+    return panel
+  }
+
+  @IBAction func eqPopUpButtonAction(_ sender: NSPopUpButton) {
+    let tag = sender.selectedTag()
+    let name = sender.titleOfSelectedItem
+    switch tag {
+    case -3: // save
+      let alert = makeProfileNameValidationAlert(isNewProfile: true)
+      let response = alert.runModal()
+      if response == .alertFirstButtonReturn {
+        let newProfile = EQProfile(fromCurrentSliders: eqSliders)
+        userEQs[inputString] = newProfile
+        menuNeedsUpdate(eqPopUpButton.menu!)
+        eqPopUpButton.selectItem(withTitle: inputString)
+      } else {
+        eqPopUpButton.selectItem(withTitle: lastUsedProfileName)
+      }
+    case -2: // rename
+      let alert = makeProfileNameValidationAlert(isNewProfile: false)
+      let response = alert.runModal()
+      if response == .alertFirstButtonReturn {
+        let profile = userEQs.removeValue(forKey: lastUsedProfileName)
+        userEQs[inputString] = profile
+        menuNeedsUpdate(eqPopUpButton.menu!)
+        eqPopUpButton.selectItem(withTitle: inputString)
+      } else {
+        eqPopUpButton.selectItem(withTitle: lastUsedProfileName)
+      }
+    case -1: // delete
+      userEQs.removeValue(forKey: lastUsedProfileName)
+      menuNeedsUpdate(eqPopUpButton.menu!)
+      eqPopUpButton.selectItem(withTag: 1000)
```

Comment:
You use "remove the current EQ" in the UI. My intuition is that it will also reset the EQ to default. However it only deletes the current preset. I suggest more precise working like "Save the current EQ as preset" "Delete the current EQ preset".